apps/web/app/api/v1/management/surveys/[surveyId]/route.ts

```diff
@@ -0,0 +1,106 @@
+import { responses } from "@/lib/api/response";
+import { DatabaseError, InvalidInputError, ResourceNotFoundError } from "@formbricks/types/v1/errors";
+import { NextResponse } from "next/server";
+import { getSurvey, updateSurvey, deleteSurvey } from "@formbricks/lib/services/survey";
+import { TSurvey, ZSurveyInput } from "@formbricks/types/v1/surveys";
+import { transformErrorToDetails } from "@/lib/api/validator";
+import { hasUserEnvironmentAccess } from "@/lib/api/apiHelper";
+import { authenticateRequest } from "@/app/api/v1/auth";
+
+async function fetchAndValidateSurvey(authentication: any, surveyId: string): Promise<TSurvey | null> {
+  const survey = await getSurvey(surveyId);
+  if (!survey) {
+    return null;
+  }
+  if (!(await canUserAccessSurvey(authentication, survey))) {
+    throw new Error("Unauthorized");
+  }
+  return survey;
+}
+
+const canUserAccessSurvey = async (authentication: any, survey: TSurvey): Promise<boolean> => {
+  if (!authentication) return false;
+
+  if (authentication.type === "session") {
+    return await hasUserEnvironmentAccess(authentication.session.user, survey.environmentId);
+  } else if (authentication.type === "apiKey") {
+    return survey.environmentId === authentication.environmentId;
```

Comment:
I think using `survey.environmentId === authentication.environmentId` in the above function is enough instead of having a separate function